.github/workflows/stage.yml

```diff
@@ -10,25 +10,75 @@ on:
-        npm ci
-    - name: Fetch config and update version
-      run: |
-        npm run bundle-config
-        node scripts/bumpVersion.js ${{ github.event.inputs.version }}
-    - name: Create Pull Request
-      uses: peter-evans/create-pull-request@v3
-      with:
-        commit-message: ${{ github.event.inputs.version }}
-        branch: "release-${{ github.event.inputs.version }}"
-        title: "Prepare release ${{ github.event.inputs.version }}"
-        body: ''
+      - uses: actions/checkout@v2
+      - name: Use Node.js 16
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16.x
+      - name: Install dependencies
+        run: |
+          npm i -g web-ext
+          npm ci
+      - name: Fetch config and update version
+        run: |
+          npm run bundle-config
+          node scripts/bumpVersion.js ${{ github.event.inputs.version }}
+      - name: Commit config and version updates
+        uses: stefanzweifel/git-auto-commit-action@8c3ed373fb7e451dbb9706bf99ad63124e33f614
+        with:
+          commit_message: ${{ github.event.inputs.version }}
+          branch: develop
+          title: "Prepare release ${{ github.event.inputs.version }}"
+
+      - name: Generate changelog
+        id: changelog
+        uses: metcalfc/changelog-generator@c3f61b2a3db543ceee83c759ac8e48419a1a2211
+        with:
+          myToken: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+        - name: Install dependencies
+          run: |
+            npm ci
+            npm i -g web-ext
+        - name: Build
+          run: |
+            npm run bundle-config
+            npm run ${{ matrix.target }}
+            echo "VERSION=$(jq -r .version ./browsers/firefox/manifest.json)" >> $GITHUB_ENV
+        - name: Package Firefox Release
+          if: matrix.target == 'release-firefox'
+          run: |
+            cd build/firefox/release && web-ext build
+            echo "BUILT_WITH=Built with node $(node --version) and npm $(npm --version)" >> $GITHUB_ENV
+
+        - name: Create release draft
+          uses: softprops/action-gh-release@9729932bfb75c05ad1f6e3a729294e05abaa7001
+          env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          with:
+            draft: true
+            prerelease: true
+            body: ${{ steps.changelog.outputs.changelog }}\n\n${{ env.BUILT_WITH }}
+            files: |
+              ./build/firefox/release/web-ext-artifacts/*.zip
+              ./build/chrome/release/*.zip
+
+        - name: Asana Workflow
+            env:
+              ASANA_ACCESS_TOKEN: ${{ secrets.ASANA_ACCESS_TOKEN }}
+              VERSION: ${{ github.event.release.tag_name }}
+              RELEASE_URL: ${{ github.event.release.html_url }}
+            run: |
+              npm install
```

Comment:
Way do we need to run `npm install` again?

Reply:
Just oversight. I'm removing the extra installs. While checking, I noticed that we also have an `install-ci` command in package.json (see [here](https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/2e0473ba17a9314c63fdfd48aac728b26da9d57a/package.json#L10)). Should we use that instead? It rebuilds puppeteer (which we don't need) but I'm not sure about the rest. cc @jonathanKingston

Reply:
Yes I think we should use `install-ci` as this prevents a potential exploit of the build via npm install scripts.